handsontable/src/editors/dateEditor/dateEditor.js

```diff
@@ -72,7 +73,22 @@ export class DateEditor extends TextEditor {
     /**
      * Prevent recognizing clicking on datepicker as clicking outside of table.
      */
-    eventManager.addEventListener(this.datePicker, 'mousedown', event => event.stopPropagation());
+    eventManager.addEventListener(this.datePicker, 'mousedown', (event) => {
+      if (hasClass(event.target, 'pika-day')) {
+        this.hideDatepicker();
+      }
+
+      event.stopPropagation();
+    });
+
+    /**
+     * Prevent caret movement in the TEXTAREA when navigating over the date picker.
+     */
+    eventManager.addEventListener(this.TEXTAREA, 'keydown', (event) => {
+      if (['ArrowUp', 'ArrowDown', 'ArrowLeft', 'ArrowRight'].includes(event.code)) {
```

Comment:
Can I ask to cover that fix with caret movement with tests?

Reply:
Honestly, I tried adding a test for that but failed 🙄 Namely, the caret position seemed to be stuck at the same position, regardless of any keypress simulations I did. Do you maybe have any tips on how to potentially overcome this?

Reply:
I see, now. I've tried to prepare some code but without success. The `event.code` property is always `undefined` using simulated events. I'd leave it for feature improvements.